### ISSUE TITLE
Don't change setState logic in any way

### ISSUE THREAD

**Issue by smashercosmo** [shadowwalker/unstated#2]
Hey, @shadowwalker 
Thx for this fork. This is awesome that someone finally did it)
However, I don't really understand why you decided to change the behaviour of setState function.
'Unstated' tries to mimic original React's setState function behaviour as close as possible.
Unstated:
https://github.com/jamiebuilds/unstated/blob/master/src/unstated.js#L24-L33
Your implementation:
https://github.com/shadowwalker/unstated/blob/typescript/src/index.tsx#L25-L27

1. In your implementation setState callback is not invoked if nextState is falsy
2. Also in react and unstated to bail out of state update you need to return null explicitly (in your version - any falsy value)
3. Typescript types do not allow nextState to be null

P.S. In general I wouldn't recommend to make any changes to original unstated source (except type annotations), so that in future in would be easier to sync.

**Comment by shadowwalker** [comment]
Hi @smashercosmo,
 
Thanks for this issue.

The reason I did it like this is that I thought it would improve the performance if there is no update to the state.

There is no clear statement about return `null` to bail out re-render in [setState documentation](https://reactjs.org/docs/react-component.html#setstate). But it does mention this:

> setState() will always lead to a re-render unless shouldComponentUpdate() returns false.

This could be a clue that in the most cases, we should enforce a re-render, even the nextState is falsy.

I also found the source of bail out method from this [issue](https://github.com/facebook/react/issues/5015), where gaearon said:

> We allow returning null from updaters to prevent an update in 16.

I will work on this, to reduce the surprise here.
